mslice/scripting/helperfunctions.py

```diff
@@ -0,0 +1,217 @@
+            if isinstance(plot_handler, CutPlot) and (plot_handler.x_log is not True or plot_handler.y_log is not True):
+                continue
+            elif isinstance(plot_handler, SlicePlot):
+                continue
+        statements.append('import {} as {}\n'.format(package, PACKAGES[package]))
+
+    return statements
+
+
+def add_header(script_lines, plot_handler):
+    for i, statement in enumerate(header(plot_handler)):
+        script_lines.insert(i, statement)
+
+
+def add_plot_statements(script_lines, plot_handler):
+    from mslice.plotting.plot_window.slice_plot import SlicePlot
+    from mslice.plotting.plot_window.cut_plot import CutPlot
+
+    add_header(script_lines, plot_handler)
+
+    line_no = len(script_lines)
+    script_lines.insert(line_no - 1, '\n')
+    script_lines[line_no] = 'ws = mc.{}\n'.format(script_lines[line_no])
+
+    if plot_handler is not None:
+        if isinstance(plot_handler, SlicePlot):
+            add_slice_plot_statements(script_lines, plot_handler)
+            add_overplot_statements(script_lines, plot_handler)
+        elif isinstance(plot_handler, CutPlot):
+            add_cut_plot_statements(script_lines, plot_handler)
+
+        script_lines.append('mc.Show()\n')
+
+    return script_lines
+
+
+def add_slice_plot_statements(script_lines, plot_handler):
+    slice = plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name]
+    momentum_axis = str(slice.momentum_axis)
+    energy_axis = str(slice.energy_axis)
+    norm = slice.norm.vmin == 0.0 and slice.norm.vmax == 1.0
+
+    script_lines.append('slice_ws = mc.Slice(ws, Axis1=\'{}\', Axis2=\'{}\', NormToOne={})\n\n'.format(
+        momentum_axis, energy_axis, norm))
+    script_lines.append('fig = plt.gcf()\n')
+    script_lines.append('ax = fig.add_subplot(111, projection=\'mslice\')\n')
+
+    if plot_handler.default_options['intensity'] is True:
+        intensity = _function_to_intensity[plot_handler.default_options['intensity_method']]
+        if plot_handler.default_options['temp_dependent']:
+            script_lines.append(
+                'mesh = ax.pcolormesh(slice_ws, cmap=\'{}\', intensity=\'{}\', temperature={})\n'.format(
+                    plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name].colourmap, intensity,
+                    plot_handler.default_options['temp']))
+        else:
+            script_lines.append(
+                'mesh = ax.pcolormesh(slice_ws, cmap=\'{}\', intensity=\'{}\')\n'.format(
+                    plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name].colourmap, intensity))
+    else:
+        script_lines.append(
+            'mesh = ax.pcolormesh(slice_ws, cmap=\'{}\')\n'.format(
+                plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name].colourmap))
+
+    script_lines.append('mesh.set_clim({}, {})\n'.format(*plot_handler.colorbar_range))
+    script_lines.append(
```

Comment:
It is not always best to opt for the most succinct way expressing things in code. This line is a little difficult to parse due to the multiple uses of the ternary operator. I think something like
```python
if plot_handler.colorbar_log:
    min, max = plot_handler.colorbar_range[0], plot_handler.colorbar_range[1]
    min = max(min, LOG_SCALE_MIN):
```
would be more readable.